crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -2010,8 +2010,9 @@ impl<TX: DbTx> BlockReader for DatabaseProvider<TX> {
                 return Ok(Some(Vec::new()))
             }
 
-            let ommers = self.tx.get::<tables::BlockOmmers>(number)?.map(|o| o.ommers);
-            return Ok(ommers)
+            let ommers =
+                self.tx.get::<tables::BlockOmmers>(number)?.map(|o| o.ommers).unwrap_or_default();
+            return Ok(Some(ommers))
```

Comment:
I was having issues with the `ommers` for the cases where the random block generated contained no ommers. This would cause the `ommers` field of the block to be `[]` while querying `ommers` would return `None` because the database didn't contain ommers for the block (see [here](https://github.com/paradigmxyz/reth/blob/3373670723896539571541442216796d7133a3ff/crates/storage/provider/src/providers/database/provider.rs#L3462) for reason). 

I think `unwrapping` here makes sense as you should only return `None` if the block doesn't exist (as far as the documentation goes at least).

Reply:
> you should only return `None` if the block doesn't exist

AFAIK we can't distinguish between a block not existing and having no ommers without an additional lookup into headers table. Your change will always return `Some([])` even if ommers don't exist for the block, and it's an incorrect behavior.

So I think we should revert this change to the provider and instead manually set the ommers field in tests.

Reply:
reverted but added docs, as I think the docs provided on the trait definition don't capture this effect.

Reply:
brilliant, appreciate it